Software/Nucleo-F446RE-DS18B20/DS18B20/ds18b20.c

```diff
@@ -153,14 +264,24 @@ void ds18b20_demo_multi( void ) {
 
         if ( result ) {
           my_printf( "DS18B20 No %i has Temp of %d \n", (i + 1), sensor_1.temperature );
+          temperature[i] = sensor_1.temperature;
         } else {
           my_printf( "Device not of type DS18B20 \n" );
         }
 
         /** end of looping through local buffer **/
       }
     }
-
+	my_printf(	"tick,%d,"
+				"DS18B20_1,%d,GradC,"
+				"DS18B20_2,%d,GradC,"
+				"DS18B20_3,%d,GradC,"
+				"DS18B20_4,%d,GradC\n",
+				tickNow,
+				temperature[0],
+				temperature[1],
+				temperature[2],
+				temperature[3]);
     my_printf( "\n" );
```

Comment:
find a way to not have printf in a library function but outside in the main-loop() level